src/Language/PureScript/CodeGen/JS/Optimizer/Inliner.hs

```diff
@@ -246,71 +239,130 @@ inlineFnComposition :: (Applicative m, MonadSupply m) => JS -> m JS
+ordInt = [(C.prelude, C.ordInt), (C.dataOrd, C.ordInt)]
+
+ordString :: [(String, String)]
+ordString = [(C.prelude, C.ordString), (C.dataOrd, C.ordString)]
+
+ordChar :: [(String, String)]
+ordChar = [(C.prelude, C.ordChar), (C.dataOrd, C.ordChar)]
+
+semigroupString :: [(String, String)]
+semigroupString = [(C.prelude, C.semigroupString), (C.dataSemigroup, C.semigroupString)]
 
-isDict :: (String, String) -> JS -> Bool
-isDict (moduleName, dictName) (JSAccessor x (JSVar y)) = x == dictName && y == moduleName
-isDict _ _ = False
+boundedBoolean :: [(String, String)]
+boundedBoolean = [(C.prelude, C.boundedBoolean), (C.dataBounded, C.boundedBoolean)]
 
-isFn :: (String, String) -> JS -> Bool
-isFn (moduleName, fnName) (JSAccessor x (JSVar y)) = x == fnName && y == moduleName
-isFn (moduleName, fnName) (JSIndexer (JSStringLiteral x) (JSVar y)) = x == fnName && y == moduleName
-isFn _ _ = False
+booleanAlgebraBoolean :: [(String, String)]
+booleanAlgebraBoolean = [(C.prelude, C.booleanAlgebraBoolean), (C.dataBooleanAlgebra, C.booleanAlgebraBoolean)]
 
-isPreludeFn :: String -> JS -> Bool
-isPreludeFn fnName = isFn (C.prelude, fnName)
+semigroupoidFn :: [(String, String)]
+semigroupoidFn = [(C.prelude, C.semigroupoidFn), (C.controlSemigroupoid, C.semigroupoidFn)]
 
-semiringNumber :: (String, String)
-semiringNumber = (C.prelude, C.semiringNumber)
+opAdd :: [(String, String)]
+opAdd = [(C.prelude, (C.+)), (C.prelude, C.add), (C.dataSemiring, C.add)]
 
-semiringInt :: (String, String)
-semiringInt = (C.prelude, C.semiringInt)
+opMul :: [(String, String)]
+opMul = [(C.prelude, (C.*)), (C.prelude, C.mul), (C.dataSemiring, C.mul)]
 
-ringNumber :: (String, String)
-ringNumber = (C.prelude, C.ringNumber)
+opEq :: [(String, String)]
+opEq = [(C.prelude, (C.==)), (C.prelude, C.eq), (C.dataEq, C.eq)]
 
-ringInt :: (String, String)
-ringInt = (C.prelude, C.ringInt)
+opNotEq :: [(String, String)]
+opNotEq = [(C.prelude, (C./=)), (C.dataEq, C.notEq)]
 
-moduloSemiringNumber :: (String, String)
-moduloSemiringNumber = (C.prelude, C.moduloSemiringNumber)
+opLessThan :: [(String, String)]
+opLessThan = [(C.prelude, (C.<)), (C.dataOrd, C.lessThan)]
 
-moduloSemiringInt :: (String, String)
-moduloSemiringInt = (C.prelude, C.moduloSemiringInt)
+opLessThanOrEq :: [(String, String)]
+opLessThanOrEq = [(C.prelude, (C.<=)), (C.dataOrd, C.lessThanOrEq)]
 
-eqNumber :: (String, String)
-eqNumber = (C.prelude, C.eqNumber)
+opGreaterThan :: [(String, String)]
+opGreaterThan = [(C.prelude, (C.>)), (C.dataOrd, C.greaterThan)]
 
-eqInt :: (String, String)
-eqInt = (C.prelude, C.eqInt)
+opGreaterThanOrEq :: [(String, String)]
+opGreaterThanOrEq = [(C.prelude, (C.>=)), (C.dataOrd, C.greaterThanOrEq)]
 
-eqString :: (String, String)
-eqString = (C.prelude, C.eqNumber)
+opAppend :: [(String, String)]
+opAppend = [(C.prelude, (C.<>)), (C.prelude, (C.++)), (C.prelude, C.append), (C.dataSemigroup, C.append)]
 
-eqBoolean :: (String, String)
-eqBoolean = (C.prelude, C.eqNumber)
```

Comment:
Gee, I wonder why the boolean and string equality inlining wasn't working :scream: